rplugin/python3/denite/ui/default.py

```diff
@@ -246,23 +254,32 @@ def update_buffer(self):
             ))
 
         del self.__vim.current.buffer[:]
-        self.__vim.current.buffer.append(
-            ['%s %s' % (
-                (self.__context['selected_icon']
-                 if i in self.__selected_candidates else ' ') +
-                (re.sub(r'([a-zA-Z])[a-zA-Z]+', r'\1', x['source'])
-                 if self.__context['short_source_names']
-                 else x['source']) if self.__is_multi else '',
-                x.get('abbr', x['word'])[:400])
-             for i, x in enumerate(self.__candidates[
-                     self.__cursor: self.__cursor + self.__winheight])])
+        self.__vim.current.buffer.append([
+            self.__get_candidate_display_text(i)
+            for i in range(self.__cursor, self.__cursor + self.__winheight)
+        ])
         del self.__vim.current.buffer[0]
         self.resize_buffer()
 
         self.__options['modified'] = False
 
         self.move_cursor()
 
+    def __get_candidate_display_text(self, index):
+        candidate = self.__candidates[index]
+        terms = []
+        if index in self.__selected_candidates:
+            terms.append(self.__selected_icon)
```

Comment:
I don't like to access a member of dictionary directly so I add a new private property. Let me know if you don't like this way.